crypto/include/internal/cryptlib_int.h

```diff
@@ -12,8 +12,9 @@
 
 /* This file is not scanned by mkdef.pl, whereas cryptlib.h is */
 
```

Comment:
I just realised that these functions are completely undocumented.  That's not all right, and I'm sorry I didn't catch on to that earlier...

Please do document, `doc/internal` is there for exactly that purpose.  I don't expect that to happen in *this* PR, I rather expect to see another PR specifically for documentation.

Reply:
I added some documentation for the functions added or changed by this PR. Once this PR is in I'll extend it in a second PR to document the other thread related functions.